### PR TITLE
Camera localizer and external usage

### PR DESCRIPTION
**Pull request by fabiencastan**
- cmake fixes
  - for external compiling
  - renaming `OpenMVG_BUILD_VOCTREE` --> `OpenMVG_USE_BOOST`
  - rig module dependency fix
- Add image matches in localization

### CODE REVIEW

File: src/cmakeFindModules/FindOpenMVG.cmake [hunk 1]
```diff
@@ -29,32 +29,46 @@ ELSE()
   MESSAGE(STATUS "NOT FOUND")
 ENDIF (OPENMVG_INCLUDE_DIR)
 
-SET(OPENMVG_LIBRARIES_NAMES  
-  openMVG_numeric
-  openMVG_system
+SET(OPENMVG_LIBRARIES_NAMES
```
Comment (pmoulon) [inline src/cmakeFindModules/FindOpenMVG.cmake]: Why did you change the order?

Comment (fabiencastan) [inline src/cmakeFindModules/FindOpenMVG.cmake]: To have the same list in `src/CMakeLists.txt``and `FindOpenMVG.cmake`.
We have just commented `openMVG_linearProgramming` because it creates problems and we have not checked why.

File: src/openMVG/voctree/database.hpp [hunk 2]
```diff
@@ -34,6 +35,14 @@ struct DocMatch
   {
     return score < other.score;
   }
+
```
Comment (simogasp) [inline src/openMVG/voctree/database.hpp]:
```c++   
bool operator==(const DocMatch& other) const
{		    
  return (score == other.score) && (id == other.score);
}		    
bool operator!=(const DocMatch& other) const
{		    
  return !(this == other);
}	
```

File: src/openMVG/localization/LocalizationResult_test.cpp [hunk 3]
```diff
@@ -48,8 +48,12 @@ localization::LocalizationResult generateRandomResult(std::size_t numPts)
...
   // random valid
   const bool valid = (numInliers % 2 == 0);
-  //
-  return localization::LocalizationResult(data, indMatch3D2D, pose, intrinsics, valid);
+
+  std::vector<voctree::DocMatch> matchedImages;
+  matchedImages.push_back(voctree::DocMatch(2, 0.5));
+  matchedImages.push_back(voctree::DocMatch(3, 0.8));
+
+  return localization::LocalizationResult(data, indMatch3D2D, pose, intrinsics, matchedImages, valid);
```
Comment (simogasp) [inline src/openMVG/localization/LocalizationResult_test.cpp]: need to add in the 2 tests the comparison between the values

in `LoadSaveBinSingle`
```c++
  // same matchedImages
  EXPECT_TRUE(res.getMatchedImages().size() == check.getMatchedImages().size());
  const auto matchedImagesGT = res.getMatchedImages();
  const auto matchedImages = check.getMatchedImages();
  for(std::size_t i = 0; i < res.getMatchedImages().size(); ++i)
  {
    EXPECT_TRUE(matchedImagesGT[i] == matchedImages[i]);
  }
```
and the same cut and paste in `LoadSaveBinVector `

Comment (simogasp) [inline src/openMVG/localization/LocalizationResult_test.cpp]: for that we need the operator `==` and `!=` for `DocMatch`. See comment in other file

File: src/openMVG/voctree/database.hpp (hunk 2 above)
Comment (fabiencastan) [inline src/openMVG/voctree/database.hpp]: It's the same than the default implementation, so we can keep the default one.

File: src/openMVG/features/feature.hpp [hunk 4]
```diff
@@ -83,6 +83,11 @@ class SIOPointFeature : public PointFeature {
   float orientation() const { return _orientation; }
   float& orientation() { return _orientation; }
 
+  Vec2f getOrientationVector() const
```
Comment (simogasp) [inline src/openMVG/features/feature.hpp]: maybe call this `getScaledOrientationVector()` and have a `getOrientationVector()` that returns only the unit vector of the orientation:
```c++
Vec2f getOrientationVector() const
{
  return Vec2f(std::cos(orientation()), std::sin(orientation()));
}
Vec2f getScaledOrientationVector() const
{
  return scale()*getOrientationVector();
}
```

File: src/openMVG/features/svgVisualization.cpp [hunk 5]
```diff
@@ -81,10 +81,9 @@ void saveKeypoints2SVG(const std::string &inputImagePath,
     if(richKeypoint)
     {
       // compute the coordinate of the point on the circle used to draw the orientation line
-      const float pointX = kpt.x()+std::cos(kpt.orientation())*kpt.scale()*radius;
-      const float pointY = kpt.y()+std::sin(kpt.orientation())*kpt.scale()*radius;
+      const Vec2f point = kpt.coords() + kpt.getOrientationVector() * radius;
```
Comment (simogasp) [inline src/openMVG/features/svgVisualization.cpp]: in case, this becomes `getScaledOrientationVector()`

File: src/openMVG/exif/sensor_width_database/ParseDatabase.hpp [hunk 6]
```diff
@@ -13,43 +12,43 @@
...
 
+namespace openMVG {
+namespace exif {
+namespace sensordb {
+
 // Parse the database
-bool parseDatabase( const std::string& sfileDatabase, std::vector<Datasheet>& vec_database )
+inline bool parseDatabase( const std::string& sfileDatabase, std::vector<Datasheet>& vec_database )
 {
   std::ifstream iFilein( sfileDatabase.c_str() );
```
Comment (simogasp) [inline src/openMVG/exif/sensor_width_database/ParseDatabase.hpp]: `.c_str()` not necessary
